tests/manager/manager-ctor-auth_mechanism-error-001.phpt

```diff
@@ -0,0 +1,42 @@
+--TEST--
+MongoDB\Driver\Manager::__construct(): authentication options are validated
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo throws(function() {
+    new MongoDB\Driver\Manager('mongodb://localhost:27017/?authMechanism=GSSAPI&authSource=admin');
```

Comment:
I think we should repeat this for MONGODB-X509 as well, just to cover all of our bases. Likewise for the URI options array variation below.